src/DynamoCoreWpf/Utilities/CompactBubbleHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using Dynamo.Extensions;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Properties;
+using ProtoCore.Mirror;
+
+namespace Dynamo.Wpf.Utilities
+{
+    /// <summary>
+    /// Helper class to process node output
+    /// </summary>
+    public static class CompactBubbleHandler
+    {
+        private static int items;
+
+        /// <summary>
+        /// Creates an instance of <cref name="CompactBubbleViewModel"/> class
+        /// from given node value
+        /// </summary>
+        /// <param name="value">Node value</param>
+        /// <returns>Instance of <cref name="CompactBubbleViewModel"/> class</returns>
+        public static CompactBubbleViewModel Process(MirrorData value)
+        {
+            items = 0;
+            var viewModel = ProcessThing(value, true);
+            viewModel.NumberOfItems = items;
+            return viewModel;
+        }        
+
+        private static CompactBubbleViewModel ProcessThing(MirrorData mirrorData, bool generateVm)
+        {
+            if (mirrorData == null) return generateVm ? new CompactBubbleViewModel(Resources.NullString, 0) : null;
+
+            if (mirrorData.IsCollection) return ProcessCollection(mirrorData, generateVm);
+
+            items++;
+
+            // generateVm is a flag to not create unused view models
+            if (!generateVm) return null;
+
+            var viewModel = new CompactBubbleViewModel();
+            if (mirrorData.Data == null && !mirrorData.IsNull && mirrorData.Class != null)
+            {
+                viewModel.NodeLabel = mirrorData.Class.ClassName;
```

Comment:
I am not sure about this case. The code is just relocated from PreviewControl.xaml.cs https://github.com/DynamoDS/Dynamo/pull/6602/files#diff-3f674ebde7209574e4dc8b9c6c1295f2L330